onnxruntime/test/framework/sparse_kernels_test.cc

```diff
@@ -0,0 +1,506 @@
+
+      ORT_ENFORCE(shape_shape.NumDimensions() == 1, "Shape must be a 1-dimensional tensor.");
+
+      TensorShape shape(shape_tensor.Data<int64_t>(), shape_shape.Size());
+
+      SparseTensor* output = ctx->Output(0, static_cast<size_t>(nnz), shape);
+      ORT_ENFORCE(output != nullptr);
+
+      memcpy(output->Values().MutableData<int64_t>(), values.Data<int64_t>(), nnz * sizeof(int64_t));
+      memcpy(output->Indices().MutableData<int64_t>(), indices.Data<int64_t>(), nnz * numdims * sizeof(int64_t));
+
+      return Status::OK();
+    }
+  };
+
+  static KernelDefBuilder KernelDef() {
+    KernelDefBuilder def;
+    def.SetName(SparseFromCOO::OpName())
+        .TypeConstraint("values", DataTypeImpl::GetTensorType<int64_t>())
+        .TypeConstraint("indices", DataTypeImpl::GetTensorType<int64_t>())
+        .TypeConstraint("shape", DataTypeImpl::GetTensorType<int64_t>())
+        .TypeConstraint("sparse_rep", DataTypeImpl::GetSparseTensorType<int64_t>());
+    return def;
+  }
+};
+
+// op SparseAbs
+struct SparseAbs {
+  static const std::string OpName() { return "SparseAbs"; };
+
+  // The ONNX schema for SparseAbs:
+  static ONNX_NAMESPACE::OpSchema OpSchema() {
+    ONNX_NAMESPACE::OpSchema schema;
+    schema.SetDoc(R"DOC(
+This operator applies the Abs op element-wise to the input sparse-tensor.
+)DOC")
+        .Input(
+            0,
+            "input",
+            "Input sparse tensor",
+            "T",
+            OpSchema::Single)
+        .Output(
+            0,
+            "output",
+            "Output sparse tensor",
+            "T",
+            OpSchema::Single)
+        .TypeConstraint(
+            "T",
+            {"sparse_tensor(int64)"},
+            "Input and Output type");
+    return schema;
+  }
+
+  // A kernel implementation of SparseAbs:
+  class OpKernelImpl final : public OpKernel {
+   public:
+    OpKernelImpl(const OpKernelInfo& info) : OpKernel{info} {}
+
+    Status Compute(OpKernelContext* ctx) const override {
+      ORT_ENFORCE(ctx->InputCount() == 1, "Expecting 1 input");
+
+      const SparseTensor* input = ctx->Input<SparseTensor>(0);
+      auto* input_values = input->Values().Data<int64_t>();
+      auto nnz = input->NumValues();
+      auto& shape = input->Shape();
+
+      // Allocate/get output-tensor:
+      SparseTensor* output = ctx->Output(0, nnz, shape);
+
+      // compute output values:
+      auto* output_values = output->Values().MutableData<int64_t>();
+      for (size_t i = 0; i < nnz; ++i)
+        output_values[i] = std::abs(input_values[i]);
+
+      // Currently, there is no way to share the indices/shape between two sparse-tensors.
+      // So, we copy indices/shape from input to output.
+      // TODO: Extend allocation-planner to enable such sharing.
+      const auto& input_indices = input->Indices();
+      memcpy(output->Indices().MutableData<int64_t>(), input_indices.Data<int64_t>(), input_indices.Size());
+
+      output->Shape() = shape;
+
```

Comment:
Should the call to ctx->Output where you pass in the shape have set this?

Reply:
You are right, thanks for catching this. It's an outdated line of code.